### PR TITLE
Fix: make cloud_credentials data test isolated

### PR DESCRIPTION
**Pull request by GiliFaroEnv0**
Issue & Steps to Reproduce / Feature Request
since we run our tests in parallel other tests that create `env0_aws_credentials` resources interfere with this test

Solution
Isolate the test and only use resources from the same test

### CODE REVIEW

File: tests/integration/024_cloud_credentials/main.tf [hunk 1]
```diff
@@ -19,9 +24,12 @@ data "env0_cloud_credentials" "aws_credentials" {
...
 
-data "env0_aws_credentials" "aws_credentials" {
-  for_each = toset(data.env0_cloud_credentials.aws_credentials.names)
-  name     = each.value
+data "env0_aws_credentials" "aws_credentials1" {
+  name     = data.env0_cloud_credentials.aws_credentials.names[index(data.env0_cloud_credentials.aws_credentials.names, env0_aws_credentials.aws_cred1.name)]
+}
+
+data "env0_aws_credentials" "aws_credentials2" {
+  name     = data.env0_cloud_credentials.aws_credentials.names[index(data.env0_cloud_credentials.aws_credentials.names, env0_aws_credentials.aws_cred2.name)]
```
Comment (liranfarage89) [inline tests/integration/024_cloud_credentials/main.tf:32]: First time seeing this format. does it work like `for_each`?

Comment (GiliFaroEnv0) [reply tests/integration/024_cloud_credentials/main.tf]: nope its the opposite ,
you take a list and search in the list for a specific value

Comment (GiliFaroEnv0) [reply tests/integration/024_cloud_credentials/main.tf]: @liranfarage89 same as https://github.com/env0/terraform-provider-env0/pull/697

**Review by liranfarage89 — APPROVED** [review]
LGTM

File: tests/integration/024_cloud_credentials/main.tf [hunk 2]
```diff
@@ -4,26 +4,35 @@ resource "random_string" "random" {
...
 
-data "env0_aws_credentials" "aws_credentials" {
-  for_each = toset(data.env0_cloud_credentials.aws_credentials.names)
-  name     = each.value
+data "env0_aws_credentials" "aws_credentials1" {
+  name     = data.env0_cloud_credentials.all_aws_credentials.names[index(data.env0_cloud_credentials.all_aws_credentials.names, env0_aws_credentials.aws_cred1.name)]
+}
+
+data "env0_aws_credentials" "aws_credentials2" {
+  name     = data.env0_cloud_credentials.all_aws_credentials.names[index(data.env0_cloud_credentials.all_aws_credentials.names, env0_aws_credentials.aws_cred2.name)]
```
Comment (sagilaufer1992) [inline tests/integration/024_cloud_credentials/main.tf:33]: why was this part added? it's not used

Comment (GiliFaroEnv0) [reply tests/integration/024_cloud_credentials/main.tf:33]: this verify you found the resource you created before,
otherwise it will fail

Comment (sagilaufer1992) [reply tests/integration/024_cloud_credentials/main.tf:33]: `aws_credentials2` is not used. in the output we only output `1`, not even `GCP`, but I get why to create `GCP` too. what's the use of `aws2`?

Comment (sagilaufer1992) [reply tests/integration/024_cloud_credentials/main.tf:33]: ok, got it, thanks
